src/bucket/SearchableBucketList.cpp

```diff
@@ -51,6 +51,15 @@ SearchableLiveBucketListSnapshot::scanForEviction(
     auto startIter = evictionIter;
     auto scanSize = sas.evictionScanSize;
 
+    // exit scan early if there's nothing to scan. We still need to call
+    // `updateStartingEvictionIterator` before exiting early in case the
```

Comment:
Is there a test that observes this behavior (i.e. iterator updates due to bucket changes)?

Reply:
We had a test for this when eviction is enabled, "iterator resets when bucket changes". I've extended it to also include the eviction scan size of 0 case. While doing this, I also removed the exit early condition. Even when eviction is disabled, it's important that the eviction iterator remain valid at all times such that we can safely resume eviction when re-enabled. The "shortcut" path actually introduces more complexity, as we skip the scan, but still need to check for bucket spills and potentially re-write the iterator if it's been invalidated. Instead of adding more edge cases, I've removed the early exit and kept the scan as is, but added more tests to ensure we don't actually make progress when eviction scan size is 0, other than maintaining iterator validity.

Reply:
Alright, yeah, I don't think it's worth writing more complex code to handle the edge cases.